src/participant/prompts/promptBase.ts

```diff
@@ -188,6 +189,14 @@ export abstract class PromptBase<TArgs extends PromptArgsBase> {
       }
 
       if (historyItem instanceof vscode.ChatResponseTurn) {
+        if (
+          historyItem.result.errorDetails?.message ===
+          ParticipantErrorTypes.FILTERED
+        ) {
+          messages.pop();
```

Comment:
I am not a huge fan of this way of adding and then removing the message retroactively but can't think of a nicer way. 

We could be less strict with what kind of errors lead to it, right now I decided to just tackle the filtered error but maybe we don't want other error-causing messages either.

Reply:
Is this message then completely gone from the history and user won't be able to see this in their UI? If so, then maybe we can also consider trimming the message short instead of completely removing it? That won't leave the user hanging wondering how their last messages are not showing up.

Reply:
Completely ignorable if that's not what is happening (I did not test it locally, sorry)

Reply:
Nope the message will still be around in the UI, this is just a matter of what we use as our message context